src/NumberTheory/vinberg.jl

```diff
@@ -0,0 +1,336 @@
+###################################################################
+# Vinberg's algorithm
+###################################################################
+
+@doc raw"""
+    _check_v0(Q::ZZMatrix, v0::ZZMatrix) -> ZZMatrix
+
+Check if the inner product of the given `v0` is positive or generate such a `v0` if `v0==0`
+"""
+function _check_v0(Q::ZZMatrix, v0::ZZMatrix)
+  if iszero(v0) # generate `v0` by finding the negative eigenvalue `neg_ev` of `Q`. We set `v0` as a primitive generator of the eigenspace of `neg_ev`
+    V = quadratic_space(QQ,Q)
+    diag, trafo = diagonal_with_transform(V)
+    @req count(is_positive, diag) == 1 "Q must be of signature (1,n)"
+    @req count(is_zero, diag) == 0 "Q must be of signature (1,n)"
+    @req count(is_negative, diag) > 0 "Q must be of signature (1,n)"
+    i = findfirst(is_positive, diag)
+    v0 = trafo[i:i,:]
+    v0 = ZZ.(denominator(v0)*v0)
```

Comment:
@ElMerkel 
here I moved from eigenvalues to to an implementation of Gram-Schmidt orthogonalisation since I am not convinced that `Q` is diagonalisable over the rationals.